### PR TITLE
fix: MongoDB implemented DAL and MVC including tests 

### PR DESCRIPTION
**Pull request by matanamar10**
fix: MongoDB implemented DAL and MVC including tests

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# .env file
```
Comment (fran2810) [inline .env]: add gitignore you should not add .env file to git

File: mongodb/mongo_handler.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
+# mongo_handler.py
+
+import os
+from pymongo import MongoClient
+from dotenv import load_dotenv
+
+# Load environment variables from .env file
+load_dotenv()
```
Comment (fran2810) [inline mongodb/mongo_handler.py]: all the env vars should not be here read about settings with pydantic 

https://docs.pydantic.dev/latest/concepts/pydantic_settings/

File: mongodb/mongo_handler.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+MONGO_CLIENT = os.getenv('MONGO_CLIENT')
+MONGO_DB_NAME = os.getenv('MONGO_DB_NAME')
+MONGO_PATRONS_COLLECTION = os.getenv('MONGO_PATRONS_COLLECTION')
+MONGO_ITEMS_COLLECTION = os.getenv('MONGO_ITEMS_COLLECTION')
+MONGO_BILLS_COLLECTION = os.getenv('MONGO_BILLS_COLLECTION')
+
+# MongoDB client setup
+client = MongoClient(MONGO_CLIENT)
```
Comment (fran2810) [inline mongodb/mongo_handler.py]: should be inside function all the setup

File: mongodb/mongo_handler.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+from pymongo import MongoClient
+from dotenv import load_dotenv
+
+# Load environment variables from .env file
+load_dotenv()
+
+MONGO_CLIENT = os.getenv('MONGO_CLIENT')
+MONGO_DB_NAME = os.getenv('MONGO_DB_NAME')
+MONGO_PATRONS_COLLECTION = os.getenv('MONGO_PATRONS_COLLECTION')
+MONGO_ITEMS_COLLECTION = os.getenv('MONGO_ITEMS_COLLECTION')
```
Comment (fran2810) [inline mongodb/mongo_handler.py]: use mongoengine not pymongo

**Review by fran2810 — CHANGES_REQUESTED** [review]
you have big things to do learn about mongoengine, pydantic settings and add .gitignore

File: mongodb/mongodb_models/patron_model.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+# Define Patron document
+from mongoengine import Document, StringField, ReferenceField, DictField, DateTimeField
+
+from mongodb.mongo_setup import settings
+
+
+class PatronDocument(Document):
+    name = StringField(required=True, max_length=60)
+    patron_id = StringField(required=True, unique=True, regex=r'^\d{9}$')
```
Comment (fran2810) [inline mongodb/mongodb_models/patron_model.py:8]: why not use the document id and convert the parsing

File: pytestdebug.log [hunk 6]
```diff
@@ -0,0 +1,498 @@
+versions pytest-8.2.2, python-3.10.11.final.0
+invocation_dir=C:\Users\amarm\PycharmProjects\python-library-project
+cwd=C:\Users\amarm\PycharmProjects\python-library-project
+args=('test_borrowing_department.py::test_borrow_a_library_item_patron_not_exist', '--debug')
```
Comment (fran2810) [inline pytestdebug.log]: what is this add to git ignore

File: mongodb/mongodb_models/library_item_model.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
+from mongoengine import Document, StringField, BooleanField
+
+from mongodb.mongo_setup import settings
+
+
+class LibraryItemDocument(Document):
+    is_borrowed = BooleanField(default=False)
```
Comment (fran2810) [inline mongodb/mongodb_models/library_item_model.py:6]: there's no id

File: mongodb/mongodb_models/library_item_model.py [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+    title = StringField(required=True, max_length=60)
+    isbn = StringField(required=True, unique=True, regex=r'^\d{9}$')
```
Comment (fran2810) [inline mongodb/mongodb_models/library_item_model.py:8]: where the representation to filed or disk

File: mongodb/mongodb_models/bills_model.py [hunk 9]
```diff
@@ -0,0 +1,8 @@
+from mongoengine import Document, StringField, FloatField
+from mongodb.mongo_setup import settings
+
+
+class Bill(Document):
+    patron_id = StringField(required=True, unique=True, regex=r'^\d{9}$')
+    amount = FloatField(required=True)
```
Comment (fran2810) [inline mongodb/mongodb_models/bills_model.py]: do not understand the fields amount of what ?

File: mongodb/mongodb_handler.py [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+def delete_bill(patron_id: str):
+    """
+    Delete the bill for a patron.
+
+    Args:
+        patron_id (str): The ID of the patron.
+    """
+    try:
+        Bill.objects(patron_id=patron_id).delete()
+    except Exception as e:
```
Comment (fran2810) [inline mongodb/mongodb_handler.py]: do not use so generic exception in so specific function

File: mongodb/mongodb_handler.py [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+        amount (float): The new bill amount.
+    """
+    try:
+        bill = Bill.objects(patron_id=patron_id).first()
+        if bill:
+            bill.amount = amount
+            bill.save()
+        else:
+            insert_bill(patron_id, amount)
+    except Exception as e:
```
Comment (fran2810) [inline mongodb/mongodb_handler.py]: dont catch wide exception for specific function

File: mongodb/mongodb_handler.py [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+            document.save()
+        else:
+            raise Exception(f"Unknown collection: {collection_name}")
+    except Exception as e:
+        raise Exception(f"Insert document failed: {e}")
+
+
+def delete_document(collection_name, query):
+    """
+    Delete documents from the specified MongoDB collection based on the query.
```
Comment (fran2810) [inline mongodb/mongodb_handler.py]: why its needed ?

File: mongodb/mongodb_handler.py [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+            if PatronDocument.objects(patron_id=document.patron_id).first():
+                raise Exception(f"Patron with id {document.patron_id} already exists.")
+            document.save()
+        elif collection_name == 'library-items':
+            # Ensure isbn is unique in library-items collection
+            if LibraryItemDocument.objects(isbn=document.isbn).first():
+                raise Exception(f"Library item with ISBN {document.isbn} already exists.")
+            document.save()
+        else:
+            raise Exception(f"Unknown collection: {collection_name}")
```
Comment (fran2810) [inline mongodb/mongodb_handler.py]: to raise generic exception and catch the same on is weird handle it ones

File: mongodb/mongodb_handler.py [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+                patron.patron_items[isbn] = borrow_date
+            elif action == "return":
+                if isbn in patron.patron_items:
+                    del patron.patron_items[isbn]
+            patron.save()
+    except Exception as e:
+        raise Exception(f"Update library patron status failed: {e}")
+
+
+def insert_document(collection_name, document: Union[PatronDocument, LibraryItemDocument]):
```
Comment (fran2810) [inline mongodb/mongodb_handler.py]: there's no need for generic function here just insert every item individually

File: mongodb/mongodb_handler.py [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+    Args:
+        patron_id (str): The ID of the patron.
+        isbn (str): The ISBN of the library item.
+        action (str): The action to perform ('borrow' or 'return').
+        borrow_date (str, optional): The date the item was borrowed (default is None).
+    """
+    try:
+        patron = PatronDocument.objects(patron_id=patron_id).first()
+        if patron:
+            if action == "borrow":
```
Comment (fran2810) [inline mongodb/mongodb_handler.py]: action should be protected by enum

File: mongodb/mongodb_handler.py [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+        isbn (str): The ISBN of the library item.
+        is_borrowed (bool): True if the item is borrowed, False if it is returned.
+    """
+    try:
+        library_item = LibraryItemDocument.objects(isbn=isbn).first()
+        if library_item:
+            library_item.is_borrowed = is_borrowed
+            library_item.save()
+    except Exception as e:
+        raise Exception(f"Update library item status failed: {e}")
```
Comment (fran2810) [inline mongodb/mongodb_handler.py]: not good use of exception there's no need for generic exception and if you wrap exception wrap it in a new custom exception

File: dal/dal.py [hunk 17]
```diff
@@ -0,0 +1,130 @@
+from mongodb.mongodb_handler import (
+    update_item_status,
+    update_patron_items,
+    insert_document,
+    delete_document, delete_bill, update_bill, insert_bill
+)
+
+
+def update_item_dal(isbn, is_borrowed):
```
Comment (fran2810) [inline dal/dal.py]: i get it that you want to create generic layer of dal but this not creating everything it still not support   openclose  thing about more elegant way to do it maybe use of class abstraction ABC in python

File: mongodb/mongodb_handler.py [hunk 18]
```diff
@@ -0,0 +1,199 @@
...
+    try:
+        library_item = LibraryItemDocument.objects.get(isbn=isbn)
+        library_item.is_borrowed = is_borrowed
+        library_item.save()
+        return True, ""
+    except DoesNotExist:
+        return False, f"Library item with ISBN {isbn} does not exist."
+    except MultipleObjectsReturned:
+        return False, f"Multiple items found with ISBN {isbn}."
+    except ValidationError as e:
```
Comment (fran2810) [inline mongodb/mongodb_handler.py]: if you dont have a good message to write the exception just treat validation error and operation error and return the e

File: mongodb/mongodb_handler.py [hunk 19]
```diff
@@ -0,0 +1,199 @@
...
+
+    Returns:
+        Tuple[bool, str]: A tuple indicating success (True) or failure (False), and an error message if applicable.
+    """
+    try:
+        library_item = LibraryItemDocument.objects.get(isbn=isbn)
+        library_item.is_borrowed = is_borrowed
+        library_item.save()
+        return True, ""
+    except DoesNotExist:
```
Comment (fran2810) [inline mongodb/mongodb_handler.py]: dont return an error string error string wrap it with dal exception

File: controllers/management/borrowing_department.py [hunk 20]
```diff
@@ -0,0 +1,113 @@
...
+        library_item.is_borrowed = True
+        borrow_date = datetime.now()
+        update_patron_dal(patron_id, library_item.isbn, "borrow", borrow_date)
+        update_item_dal(library_item.isbn, True)
+
+        logging.info(f"Library item {library_item.title} has been borrowed by patron {patron_id}")
+
+    except DoesNotExist as e:
+        logging.error(f"Failed to borrow library item: {e}")
+        raise ValueError(e)
```
Comment (fran2810) [inline controllers/management/borrowing_department.py]: why value errr wrap it in you own exception

File: controllers/library.py [hunk 21]
```diff
@@ -0,0 +1,164 @@
...
+                    raise ValueError(error_message)
+                logging.info(
+                    f"The library item {new_library_item.title} with ISBN {new_library_item.isbn} was added to '{self.name}' library"
+                )
+        except NotUniqueError as e:
+            logging.error(f"Add library item failed: {e}")
+            raise ValueError(e)
+        except ValidationError as e:
+            logging.error(f"Validation error: {e}")
+            raise ValueError(f"Validation error: {e}")
```
Comment (fran2810) [inline controllers/library.py]: why value error create LirbraryError with original error and message as attributes

File: controllers/library.py [hunk 22]
```diff
@@ -0,0 +1,164 @@
...
+        except OperationError as e:
+            logging.error(f"Database operation error: {e}")
+            raise ValueError(f"Database operation error: {e}")
+        except Exception as e:
+            logging.error(f"Add library item failed: {e}")
```
Comment (fran2810) [inline controllers/library.py]: Theres no need for general exception here use it only in main

File: controllers/library.py [hunk 23]
```diff
@@ -0,0 +1,164 @@
...
+            for new_library_item in new_library_items:
+                if new_library_item.isbn in self.library_items.keys():
+                    raise NotUniqueError(
+                        f"The library item with ISBN {new_library_item.isbn} already exists in the library"
+                    )
+                self.library_items[new_library_item.isbn] = new_library_item
+                new_library_item_document = item_pydantic_to_mongoengine(new_library_item)
+                success, error_message = insert_document_dal('library-items', new_library_item_document)
+                if not success:
+                    raise ValueError(error_message)
```
Comment (fran2810) [inline controllers/library.py]: dont raise and catch in the same function

**Review by fran2810 — CHANGES_REQUESTED** [review]
you need to do few changes:

1. create custom error and use hierarchy in it for ease of use for catching. like DalError LibraryError and not raise key value
2. if you dont do something special just print the e and dont catch million exceptions if youre not sure what to do with them or catch tem togater like NotUniqueError, ValidationError and Operation Error
3. dont log error so many times catch exception in the main one and log it this how the message attribute an help u
4. dont catch general exceptiso many times your application must be preidctable

File: controllers/library.py [hunk 24]
```diff
@@ -0,0 +1,112 @@
...
+from models.patrons.patron import Patron
+from pydantic import BaseModel
+from typing import Dict, List
+from dal.dal import insert_document_dal, delete_document_dal
+from utils.utils import patron_pydantic_to_mongoengine, item_pydantic_to_mongoengine
+from mongoengine import DoesNotExist
+from utils.custom_errors import *
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
```
Comment (dvirgilad) [inline controllers/library.py]: Why is this here?

File: controllers/library.py [hunk 25]
```diff
@@ -0,0 +1,112 @@
...
+    def add_new_library_items_to_the_library(self, new_library_items: List[LibraryItem]):
+        """
+        Add new library items to the library system.
+
+        Args:
+            new_library_items (List[LibraryItem]): List of library items to add.
+        """
+        for new_library_item in new_library_items:
+            if new_library_item.isbn in self.library_items.keys():
+                raise NotUniqueLibraryItemError(
```
Comment (dvirgilad) [inline controllers/library.py]: rename to AlreadyExists error and make generic. You can have a PatronAlreadyExists and a ItemAlreadyExists

File: controllers/library.py [hunk 26]
```diff
@@ -0,0 +1,112 @@
...
+    def add_new_patron_to_the_library(self, patrons_to_add: List[Patron]):
+        """
+        Add new patrons to the library system.
+
+        Args:
+            patrons_to_add (List[Patron]): List of patrons to add.
+        """
+        for patron in patrons_to_add:
+            if patron.patron_id in self.patrons:
+                raise NotUniquePatronError(f"The patron with ID {patron.patron_id} already exists in the library")
```
Comment (dvirgilad) [inline controllers/library.py]: If you you know this is an error about a patron, why do you need all this text?

File: controllers/library.py [hunk 27]
```diff
@@ -0,0 +1,112 @@
...
+        """
+        Remove patrons from the library system.
+
+        Args:
+            patrons_to_remove (List[Patron]): List of patrons to remove.
+        """
+
+        for patron in patrons_to_remove:
+            if patron.patron_id not in self.patrons:
+                raise NotExistingPatron(f"The patron with ID {patron.patron_id} does not exist in the library")
```
Comment (dvirgilad) [inline controllers/library.py]: You need to rename your errors to be in past tense. For example: PatronNotFound or PatronAlreadyExists.
The same comments about the other errors also apply here

File: controllers/library.py [hunk 28]
```diff
@@ -0,0 +1,112 @@
...
+
+    def remove_library_item_from_the_library(self, library_item_isbn: str):
+        """
+        Remove an item from the library system by ISBN.
+
+        Args:
+            library_item_isbn (str): ISBN of the item to remove.
+        """
+        if library_item_isbn not in self.library_items.keys():
+            raise DoesNotExist(f"The library item with ISBN {library_item_isbn} does not exist in the library")
```
Comment (dvirgilad) [inline controllers/library.py]: why do you have a `NotExistingPatron` error but not the same error for library items?

File: controllers/management/borrowing_department.py [hunk 29]
```diff
@@ -0,0 +1,79 @@
...
+
+    Raises:
+    - ValueError: If the patron does not exist, the item is not in the library, the item is not borrowed,
+                  or if the patron has outstanding bills.
+    """
+    patron = library.patrons.get(patron_id)
+    if not patron:
+        raise NotExistingPatron(f"Patron with patron id {patron_id} was not found in the library")
+    if library_item.isbn not in library.library_items.keys():
+        raise NotExistingLibraryItem(f"The library item with ISBN {library_item.isbn} was not found in the library")
```
Comment (dvirgilad) [inline controllers/management/borrowing_department.py]: so you do have this error

File: controllers/management/borrowing_department.py [hunk 30]
```diff
@@ -0,0 +1,79 @@
...
+    if not library_item.is_borrowed:
+        raise NotBorrowedLibraryItem(f"Library item with ISBN {library_item.isbn} is not borrowed")
+
+    patron_calculated_bill = calculate_bill(patron=patron)
+    library.bills[patron_id] = patron_calculated_bill
+
+    if library.bills[patron_id] != 0:
+        update_bill_dal(patron_id, patron_calculated_bill)
+        raise ExistingBillToPay(f"Patron {patron_id} needs to pay their bill before returning library items")
```
Comment (dvirgilad) [inline controllers/management/borrowing_department.py]: You need to change how you construct errors. You know exactly what kind of error this is so why write all this text every time you raise it?

File: controllers/management/borrowing_department.py [hunk 31]
```diff
@@ -0,0 +1,79 @@
...
+    patron.remove_library_item_from_patron(library_item)
+    library_item.is_borrowed = False  # The book is not borrowed anymore.
+    update_patron_dal(patron_id, library_item.isbn, "return")
+    update_item_dal(library_item.isbn, False)
+
+    logging.info(f"Library item {library_item.title} has been returned by patron {patron_id}")
+
+
+def borrow_a_library_item(library: Library, library_item: LibraryItem, patron_id: str):
```
Comment (dvirgilad) [inline controllers/management/borrowing_department.py]: rename to `borrow_library_item`

File: dal/dal.py [hunk 32]
```diff
@@ -0,0 +1,94 @@
...
```
Comment (dvirgilad) [inline dal/dal.py]: Also, what are these function names???? 
For example, rename this to `update_item` so when you call it, it wil be `dal.update_item(...)`

File: dal/dal.py [hunk 33]
```diff
@@ -0,0 +1,94 @@
...
+    """
+    Update the status of a library item identified by its ISBN.
+
+    Args:
+        isbn (str): The ISBN of the library item.
+        is_borrowed (bool): True if the item is borrowed, False if it is returned.
+
+    """
+    update_item_status(isbn, is_borrowed)
```
Comment (dvirgilad) [inline dal/dal.py]: If the only purpose of this function is to call another function. It has no reason to exist (in this case).
It is misleading to call this function `update_item` when all it does is change status.
If I wanted to rename a book or change an author, Intuitively, I would call this function

File: dal/dal.py [hunk 34]
```diff
@@ -0,0 +1,94 @@
...
+
+
+def update_patron_dal(patron_id, isbn, action, borrow_date=None):
```
Comment (dvirgilad) [inline dal/dal.py]: Like @fran2810 said, It seems that you are trying to create an abstraction for your DAL (which is good). But this is not the way to go about it. Read about ABC (abstract base class)

File: mongodb/mongo_setup.py [hunk 35]
```diff
@@ -0,0 +1,30 @@
+from mongoengine import connect, disconnect
+from pydantic_settings import BaseSettings
+from dotenv import load_dotenv
+
+load_dotenv()
```
Comment (dvirgilad) [inline mongodb/mongo_setup.py]: If you are using BaseSettings, why do you need this?. Also, **NEVER CALL FUNCTIONS OUTSIDE OF A FUNCTION!!!!!!**

File: mongodb/mongo_setup.py [hunk 36]
```diff
@@ -0,0 +1,30 @@
...
+    mongo_db_name: str
+    mongo_patrons_collection: str
+    mongo_items_collection: str
+    mongo_bills_collection: str
+
+    class Config:
+        env_file = '.env'
+
+
+settings = MongoDBSettings()
```
Comment (dvirgilad) [inline mongodb/mongo_setup.py]: Same here, All initialization functions need to be in main

File: mongodb/mongo_setup.py [hunk 37]
```diff
@@ -0,0 +1,30 @@
+from mongoengine import connect, disconnect
+from pydantic_settings import BaseSettings
+from dotenv import load_dotenv
+
+load_dotenv()
+
+
+# Define MongoDB settings class
+class MongoDBSettings(BaseSettings):
```
Comment (dvirgilad) [inline mongodb/mongo_setup.py]: This (a class of app settings from .env) should not be here. Create a separate `env_utils.py` files to store and handle them.

File: src/main.py [hunk 38]
```diff
@@ -1,27 +1,64 @@
...
+        teacher1 = Teacher(name="One", patron_id="123123123")
+        teacher2 = Teacher(name="Two", patron_id="123123144")
+        teachers = [teacher1, teacher2]
+        my_library.add_new_library_items_to_the_library(books)
+        my_library.add_new_patron_to_the_library(students)
+        borrow_a_library_item(my_library, book1, student3.patron_id)
+        my_library.add_new_patron_to_the_library(teachers)
+        return_a_library_item(my_library, book1, student3.patron_id)
+        borrow_a_library_item(my_library, book1, student3.patron_id)
+        my_library.add_new_library_items_to_the_library(books)
```
Comment (dvirgilad) [inline src/main.py]: All this code is for testing so move it to a different file. I don't want to see this in main.

File: src/main.py [hunk 39]
```diff
@@ -1,27 +1,64 @@
...
+    except ExistingBillToPay as e:
+        logging.error(f"Outstanding bill: {e}")
+        raise
+    except NotBorrowedLibraryItem as e:
+        logging.error(f"Item not borrowed: {e}")
+        raise
+    except BorrowedLibraryItem as e:
+        logging.error(f"Item already borrowed: {e}")
+        raise
+    except Exception as e:
```
Comment (dvirgilad) [inline src/main.py:17]: What

**Review by dvirgilad — CHANGES_REQUESTED** [review]
Left several comments. @fran2810 also left several comments you have not changed yet so look into those as well

File: mongodb/mongodb_models/bills_model.py (hunk 9 above)
Comment (matanamar10) [reply mongodb/mongodb_models/bills_model.py]: amount of the total bill to pay

File: mongodb/mongodb_handler.py (hunk 10 above)
Comment (matanamar10) [reply mongodb/mongodb_handler.py]: Done

File: mongodb/mongodb_handler.py (hunk 19 above)
Comment (matanamar10) [reply mongodb/mongodb_handler.py]: done

File: controllers/management/borrowing_department.py [hunk 40]
```diff
@@ -0,0 +1,79 @@
...
+    - ValueError: If the patron does not exist, the item is not in the library, the item is not borrowed,
+                  or if the patron has outstanding bills.
+    """
+    patron = library.patrons.get(patron_id)
+    if not patron:
+        raise PatronNotFoundError(patron_id)
+    if library_item.isbn not in library.library_items.keys():
+        raise LibraryItemNotFoundError(library_item.isbn)
+    if not library_item.is_borrowed:
+        raise BorrowedLibraryItemNotFound(library_item.isbn)
```
Comment (dvirgilad) [inline controllers/management/borrowing_department.py]: This is not an indicative error since this is not what has happened here

File: env_utils.py [hunk 41]
```diff
@@ -0,0 +1,16 @@
+from dotenv import load_dotenv
+from pydantic_settings import BaseSettings
+
+load_dotenv()
```
Comment (dvirgilad) [inline env_utils.py]: Is this necessary?

File: env_utils.py [hunk 42]
```diff
@@ -0,0 +1,16 @@
...
+# Define MongoDB settings class
+class MongoDBSettings(BaseSettings):
+    mongo_client: str
+    mongo_db_name: str
+    mongo_patrons_collection: str
+    mongo_items_collection: str
+    mongo_bills_collection: str
+
+    class Config:
+        env_file = '.env'
```
Comment (dvirgilad) [inline env_utils.py]: Why do you have random python files in the root directory?
Where should these files be placed?

File: mongodb/mongo_repository.py [hunk 43]
```diff
@@ -0,0 +1,64 @@
...
+        patron = PatronDocument.objects(patron_id=patron_id).first()
+        if patron:
+            if action == 'borrow':
+                patron.patron_items[isbn] = borrow_date
+            elif action == 'return':
+                patron.patron_items.pop(isbn, None)
+            patron.save()
+
+    def insert_document(self, collection_name, document: Union[PatronDocument, LibraryItemDocument]):
+        document.save()
```
Comment (dvirgilad) [inline mongodb/mongo_repository.py:77]: Why are you passing arguments that you are not using?

File: mongodb/mongodb_models/bills_model.py (hunk 9 above)
Comment (dvirgilad) [reply mongodb/mongodb_models/bills_model.py]: rename this already

**Review by dvirgilad — COMMENTED** [review]
Looks good, keep working on MVC code structure.

File: pytestdebug.log
Comment (dvirgilad) [inline pytestdebug.log:1]: add to gitignore

File: dal/dal.py [hunk 44]
```diff
@@ -0,0 +1,120 @@
...
+    @abstractmethod
+    def update_item_status(self, isbn: str, is_borrowed: bool) -> None:
+        """
+        Update the status of a library item by its ISBN.
+
+        Args:
+            isbn (str): The ISBN of the library item.
+            is_borrowed (bool): True if the item is borrowed, False if it is returned.
+        """
+        pass
```
Comment (dvirgilad) [inline dal/dal.py:21]: why does this method exist?

**Review by dvirgilad — COMMENTED** [review]
left a few small comments
